crates/re_sdk/src/recording_stream.rs

```diff
@@ -1104,6 +1157,53 @@ impl RecordingStream {
         self.set_sink(Box::new(crate::log_sink::TcpSink::new(addr, flush_timeout)));
     }
 
+    /// Spawns a new Rerun Viewer process from an executable available in PATH, then swaps the
+    /// underlying sink for a [`crate::log_sink::TcpSink`] sink pre-configured to send data to that
+    /// new process.
+    ///
+    /// `flush_timeout` is the minimum time the [`TcpSink`][`crate::log_sink::TcpSink`] will
+    /// wait during a flush before potentially dropping data.  Note: Passing `None` here can cause a
+    /// call to `flush` to block indefinitely if a connection cannot be established.
+    ///
+    /// This is a convenience wrapper for [`Self::set_sink`] that upholds the same guarantees in
+    /// terms of data durability and ordering.
+    /// See [`Self::set_sink`] for more information.
+    pub fn spawn(
+        &self,
+        opts: &crate::SpawnOptions,
+        flush_timeout: Option<std::time::Duration>,
+    ) -> RecordingStreamResult<()> {
+        use std::{net::TcpStream, time::Duration};
+
+        if forced_sink_path().is_some() {
+            re_log::debug!("Ignored setting new TcpSink since _RERUN_FORCE_SINK is set");
+            return Ok(());
+        }
+
+        let connect_addr = opts.connect_addr();
+
+        if TcpStream::connect_timeout(&connect_addr, Duration::from_millis(1)).is_ok() {
+            re_log::info!(
+                addr = %opts.listen_addr(),
+                "A process is already listening at this address. Trying to connect instead."
+            );
+        } else {
+            crate::spawn(opts).map_err(RecordingStreamError::SpawnViewer)?;
+
+            // Give the newly spawned Rerun Viewer some time to bind.
+            for _ in 0..5 {
+                if TcpStream::connect_timeout(&connect_addr, Duration::from_millis(1)).is_ok() {
+                    break;
+                }
+                std::thread::sleep(Duration::from_millis(100));
+            }
+        }
+
+        self.connect(connect_addr, flush_timeout);
+
+        Ok(())
+    }
```

Comment:
can we avoid duplicating code?

Reply:
Do we even want this second spawn? I think we should chose the connection type once at startup (in the builder), then stick to it

Reply:
We have to be consistent: we document everywhere than you can start buffering data and decide where to send it later on, and this is how all other modes work.

I'll look into deduplicating code if possible.